core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -364,7 +364,23 @@ private[chisel3] class ChiselContext() {
 }
 
 private[chisel3] class DynamicContext(val annotationSeq: AnnotationSeq, val throwOnFirstError: Boolean) {
+  val importedDefinitionAnnos = annotationSeq.collect { case a: ImportedDefinitionAnnotation[_] => a }
+
+  // Ensure there are no repeated names for imported Definitions
+  val importedDefinitionNames = importedDefinitionAnnos.map { a => a.importedDefinition.proto.name }
+  if (importedDefinitionNames.distinct.length < importedDefinitionNames.length) {
+    throwException("Imported Definitions must have distinct names.")
```

Comment:
It would be really helpful if we could show the ones that are NOT distinct... perhaps something like:

```suggestion
  if (importedDefinitionNames.distinct.length < importedDefinitionNames.length) {
    val duplicates = importedDefinitionNames.distinct.diff(importedDefinitionNames)
    throwException(s"Imported Definitions must have distinct names, the following are duplicates: $duplicates")
```